crates/subspace-farmer/Cargo.toml

```diff
@@ -55,6 +55,7 @@ subspace-erasure-coding = { version = "0.1.0", path = "../subspace-erasure-codin
 subspace-farmer-components = { version = "0.1.0", path = "../subspace-farmer-components" }
 subspace-core-primitives = { version = "0.1.0", path = "../subspace-core-primitives" }
 subspace-kzg = { version = "0.1.0", path = "../../shared/subspace-kzg" }
+subspace-logging = { version = "0.0.1", path = "../../shared/subspace-logging" }
```

Comment:
The “deps” line changes this from an optional dependency into a required one for the binary only.

```suggestion
subspace-logging = { version = "0.0.1", path = "../../shared/subspace-logging", optional = true }
```

Reply:
is that really optional ? what's will happen if this libs is not here ? I mean this optional flag is for which purpose ?

Reply:
Optional flag means it will not include dependency by default, it'll require explicit enabling, see other dependencies with `optional = true` for examples of how this is used.